src/scheduler/job_info.c

```diff
@@ -2896,6 +2899,14 @@ find_and_preempt_jobs(status *policy, int pbs_sd, resource_resv *hjob, server_in
 				job->job->is_preempted = 1;
 				job->job->time_preempted = sinfo->server_time;
 				sinfo->num_preempted++;
+				/* We need to correct our calendar */
+				if (sinfo->calendar != NULL) {
+					te = find_timed_event(sinfo->calendar->events, job->name, TIMED_END_EVENT, 0);
```

Comment:
@prakashcv13 I faced the exact same issue yesterday while testing something else in scheduler. I was going to log a ticket for this issue but it seems that you are already onto it, Yay!!
Now, you change is fine it is just that we should do this in update_resresv_on_end() function. Because any running job that we end (that includes preempted jobs) we should remove their end event from the calendar. Please move this code to that function.

Reply:
So @bhroam and I had more discussion on this. The outcome is that we can not really delete the event in update_resresv_on_end() because that function is also called from calendaring code while performing an event. And if the even being performed turns out to be an end event then it means that we will end up working on a deleted structure.
Instead of deleting the event, we should mark the event as disabled. There is an event parameter called disabled which can be turned on to say that ignore the event. This change will also result into change in how we find timed event in the calendar. Currently "find_timed_event()" function does not ignore disabled events. We will have to pass it a flag "IGNORE_DISABLED_EVENTS" to find_timed_event function and process that just like how find_init_timed_event function does.

Reply:
@arungrover and @bhroam - how about passing a flag to update_universe_on_end() to indicate that we are calendaring events. If that flag is set, we do not delete the event. As of now, we are not using the flags parameter at all in the update_universe_on_end() function.
If there are issues in the approach I am suggesting - I will remove the flags parameter as well.
Please let me know.

Reply:
@prakashcv13 what you are proposing will surely work. I just feel it will make update_resresv_on_end special.

Reply:
@prakashcv13 I agree with @arungrover that passing a flag into update_universe_on_end() would be a special case.  We should avoid special case code.  As for the flag parameter to update_universe_on_end(), it was used.  I removed the use of it a few months ago and didn't realize that was the only use.